src/features/island/collectibles/components/pets/PetInfo.tsx

```diff
@@ -0,0 +1,96 @@
+import { SUNNYSIDE } from "assets/sunnyside";
+import { Label } from "components/ui/Label";
+import { ResizableBar } from "components/ui/ProgressBar";
+import { getObjectEntries } from "features/game/expansion/lib/utils";
+import { ITEM_DETAILS } from "features/game/types/images";
+import {
+  getExperienceToNextLevel,
+  Pet,
+  PET_CATEGORIES,
+  PetName,
+} from "features/game/types/pets";
+import levelUp from "assets/icons/level_up.png";
+import xpIcon from "assets/icons/xp.png";
+import React, { memo } from "react";
+
+interface Props {
+  petName: PetName;
+  petData: Pet;
+}
+
+const PetInfoComponent: React.FC<Props> = ({ petName, petData }) => {
+  const { level, percentage, currentProgress, experienceBetweenLevels } =
+    getExperienceToNextLevel(petData.experience);
+
+  const petImage = ITEM_DETAILS[petName].image;
+
+  // Find pet type and categories
+  const petTypeEntry = getObjectEntries(PET_CATEGORIES).find(
+    ([, petCategory]) => petCategory.pets.includes(petName),
+  );
+  const petType = petTypeEntry?.[0];
+  const petCategory = petTypeEntry?.[1];
+  return (
+    <div className="flex flex-col gap-4 p-2">
+      {/* Pet Header */}
+      <div className="flex flex-col sm:flex-row items-center gap-4">
+        <img
+          src={petImage}
+          alt={petName}
+          className="w-24 h-24 object-contain"
+        />
+        <div className="flex-1">
+          <Label type="default" className="text-lg mb-2">
+            {petName}
+          </Label>
+
+          {/* Pet Type and Categories */}
+          {petType && petCategory && (
+            <div className="flex flex-wrap gap-2 mb-3">
+              <Label type="info" className="text-xs">
+                {petType}
+              </Label>
+              <Label type="warning" className="text-xs">
```

Comment:
For consistency, we reserve the yellow label for reward/prize type labels.

For pet types, consider just using a single colour for both. `formula` or `chill` could work here